lib/mix/lib/mix/tasks/new.ex

```diff
@@ -158,7 +158,7 @@ defmodule Mix.Tasks.New do
   end
 
   defp check_application_name!(name, inferred?) do
-    unless name =~ ~r/^[a-z][\w_]*$/ do
+    unless name =~ ~r/^[a-z][a-z0-9_]*$/ do
       Mix.raise "Application name must start with a letter and have only lowercase " <>
                 "letters, numbers and underscore, got: #{inspect name}" <>
```

Comment:
I thin the use of English is not quite right,
it is saying that is **must have letter and numbers and underscores.**
I would rephrase it like this:
"Application name can only contain lowercase letters, numbers and underscores, and it must start with a letter"